tsl/src/compression/compression.c

```diff
@@ -1761,11 +1623,34 @@ row_decompressor_decompress_row(RowDecompressor *decompressor)
 			HeapTuple decompressed_tuple = heap_form_tuple(decompressor->out_desc,
 														   decompressor->decompressed_datums,
 														   decompressor->decompressed_is_nulls);
-			heap_insert(decompressor->out_rel,
-						decompressed_tuple,
-						decompressor->mycid,
-						0 /*=options*/,
-						decompressor->bistate);
+			if (tuplesortstate == NULL)
+			{
+				heap_insert(decompressor->out_rel,
+							decompressed_tuple,
+							decompressor->mycid,
+							0 /*=options*/,
+							decompressor->bistate);
+			}
+			else
+			{
+				TupleTableSlot *slot =
```

Comment:
I wonder if the re-creation of the slot is necessary. Couldn't you create it once and reuse it in the loop here?